RecoVertex/BeamSpotProducer/plugins/BeamSpotOnlineProducer.cc

```diff
@@ -129,25 +161,59 @@ bool BeamSpotOnlineProducer::shouldShout(const edm::Event& iEvent) const {
   return true;  // Default to "shout" if the record is missing
 }
 
-bool BeamSpotOnlineProducer::processTransientRecord(const edm::EventSetup& iSetup,
-                                                    reco::BeamSpot& result,
-                                                    bool shoutMODE) {
-  auto const& spotDB = iSetup.getData(beamTransientToken_);
+bool BeamSpotOnlineProducer::processRecords(const edm::Event& iEvent,
+                                            const edm::EventSetup& iSetup,
+                                            reco::BeamSpot& result,
+                                            bool shoutMODE) {
+  auto const spotDBLegacy = getBeamSpotFromRecord(beamTokenLegacy_, iEvent, iSetup);
+  auto const spotDBHLT = getBeamSpotFromRecord(beamTokenHLT_, iEvent, iSetup);
```

Comment:
Are the `BeamSpotOnlineObjects` objects copied (as opposed to passed by const reference) on purpose?

Reply:
well, I thought there were no difference. I will change it to const&

Reply:
changed